botocore/client.py

```diff
@@ -208,6 +209,10 @@ def _get_client_args(self, service_model, region_name, is_secure,
         self._inject_s3_configuration(
             config_kwargs, scoped_config, client_config)
         self._conditionally_unregister_fix_s3_host(endpoint_url, event_emitter)
+        if service_name == 's3':
+            # We want to make double sure that we don't create this object
```

Comment:
Probably is a bit extreme for this change. But what if we created specific subclasses for the ``ClientCreator`` and/or ``BaseClient`` like ``S3ClientCreator`` and/or ``S3Client``? That would remove the issue of having specific if statements, registering unused handlers, and instantiating unused objects.